backends/graphics3d/android/android-graphics3d.cpp

```diff
@@ -0,0 +1,982 @@
+	assert(surface->getPixels());
+
+	return surface;
+}
+
+void AndroidGraphics3dManager::unlockScreen() {
+	ENTER();
+
+	GLTHREADCHECK;
+
+	assert(_game_texture->dirty());
+}
+
+void AndroidGraphics3dManager::fillScreen(uint32 col) {
+	ENTER("%u", col);
+
+	GLTHREADCHECK;
+
+	_game_texture->fillBuffer(col);
+}
+
+void AndroidGraphics3dManager::copyRectToScreen(const void *buf, int pitch,
+										int x, int y, int w, int h) {
+	ENTER("%p, %d, %d, %d, %d, %d", buf, pitch, x, y, w, h);
+
+	GLTHREADCHECK;
+
+	_game_texture->updateBuffer(x, y, w, h, buf, pitch);
+}
+
+void AndroidGraphics3dManager::initSize(uint width, uint height,
+					const Graphics::PixelFormat *format) {
+	initViewport();
+
+	// resize game texture
+	initSizeIntern(width, height, 0);
+
+	_game_texture->setGameTexture();
+	// format is not used by the gfx_opengl driver, use fake format
+	_game_pbuf.set(Graphics::PixelFormat(), 0);
+}
+
+void AndroidGraphics3dManager::initSizeIntern(uint width, uint height,
+								const Graphics::PixelFormat *format) {
+	ENTER("%d, %d, %p", width, height, format);
+
+	bool engineSupportsArbitraryResolutions = !g_engine || g_engine->hasFeature(Engine::kSupportsArbitraryResolutions);
+	if (engineSupportsArbitraryResolutions) {
+		width = JNI::egl_surface_width;
+		height = JNI::egl_surface_height;
+	}
+
+	GLTHREADCHECK;
+
+#ifdef USE_RGB_COLOR
+	initTexture(&_game_texture, width, height, format);
+#else
+	_game_texture->allocBuffer(width, height);
+#endif
+#ifdef USE_GLES2
+	delete _frame_buffer;
+	_frame_buffer = new OpenGL::FrameBuffer(_game_texture->getTextureName(), _game_texture->width(), _game_texture->height(), _game_texture->texWidth(), _game_texture->texHeight());
+	_frame_buffer->attach();
+#endif
+
+	updateScreenRect();
+	updateEventScale();
+
+	// Don't know mouse size yet - it gets reallocated in
+	// setMouseCursor.  We need the palette allocated before
+	// setMouseCursor however, so just take a guess at the desired
+	// size (it's small).
+	_mouse_texture_palette->allocBuffer(20, 20);
+
+	clearScreen(kClear);
+}
+
+int AndroidGraphics3dManager::getScreenChangeID() const {
+	return _screenChangeID;
+}
+
+bool AndroidGraphics3dManager::showMouse(bool visible) {
+	ENTER("%d", visible);
+
+	_show_mouse = visible;
+
+	return true;
+}
+
+void AndroidGraphics3dManager::warpMouse(int x, int y) {
+	ENTER("%d, %d", x, y);
+
+	Common::Event e;
+
+	e.type = Common::EVENT_MOUSEMOVE;
+	e.mouse.x = x;
+	e.mouse.y = y;
+
+	clipMouse(e.mouse);
+
+	setMousePosition(e.mouse.x, e.mouse.y);
+
+	dynamic_cast<OSystem_Android *>(g_system)->pushEvent(e);
+}
+
+void AndroidGraphics3dManager::setMouseCursor(const void *buf, uint w, uint h,
+										int hotspotX, int hotspotY,
+										uint32 keycolor, bool dontScale,
+										const Graphics::PixelFormat *format) {
+	ENTER("%p, %u, %u, %d, %d, %u, %d, %p", buf, w, h, hotspotX, hotspotY,
+			keycolor, dontScale, format);
+
+	GLTHREADCHECK;
+
+#ifdef USE_RGB_COLOR
```

Comment:
I think it was here in android3d because the backend could be used with 2D games.
In this PR this graphical backend is only for 3D so I think it could be removed.
It depends on if we want to reuse this code later to merge both graphical backends.

Reply:
I think it was copy and paste from 2d backend

Reply:
On a second thought, it is needed if someone wants to build with configure option `--disable-16bit`. In this case the `USE_RGB_COLOR` is undefined and some functions are unavailable.

Reply:
so it's rather build thing, not runtime, since no such case

Reply:
Yes indeed